src/test/common/clara/tools/test_fact_graph.cljc

```diff
@@ -0,0 +1,141 @@
+(ns clara.tools.test-fact-graph
+  (:require
+    #?(:clj [clara.tools.testing-utils :refer [def-rules-test
+                                               side-effect-holder] :as tu])
+    #?(:cljs [clara.tools.testing-utils :refer [side-effect-holder]
+              :refer-macros [def-rules-test] :as tu])
```

Comment:
In a more modern cljs version ([v1.9.198 specifically](https://github.com/clojure/clojurescript/blob/master/changes.md#19198))This namespace shouldn't have to distinguish between `:refer-macros` and `:refer`. However, I think that the ns `clara.tools.testing-utils` needs to require it's own macros though in it's `:cljs` branch though to allow for this implicit macro inference to kick in (a practice we should use in all relevant places in this codebase).
e.g.
```clj
:cljs
(ns clara.tools.testing-utils
  (:require [clara.rules.update-cache.core :as uc])
  (:require-macros clara.tools.testing-utils))
```

Since Clara is still supporting an old version of CLJS (see #363 ), I think we should stick to using ns aliases to refer to all symbols. Macro inference in CLJS has been supported for a long time for aliases I believe.

So:
```clj
#?(:cljs [clara.tools.testing-utils :as tu])

;; later
(tu/def-rules-test <etc>)
```

This applies to the `clojure.test` vs `cljs.test` stuff too, and maybe more places.

I think these `ns` declarations are getting a bit heavy and just trying to simplify before things get too messy.

This is mostly a suggestion though. If you are strongly opposed to doing it in this PR, I could understand that.